File: newterm/lab0/ex_2.py
class stack:
    def __init__(self):
        self.a = []
    def push(self, input):
        self.a += [input]
    def pop(self):
        hold = self.a[len(self.a) - 1]
        self.a = self.a[:-1]
        return(hold)


stack = stack()
stack.push(1)
stack.push(2)
stack.push(3)
print(stack.pop())
stack.push(4)
print(stack.pop())
print(stack.pop())
print(stack.pop())

File: newterm/lab0/ex_2.c
struct stack {
     char value;
     struct stack *next;
  };
  typedef struct stack stack;

  int push(stack **x, char value){
    stack *newHead;
    if (x == NULL){return -1;}
    newHead = (stack *)malloc(sizeof(stack));
    if(newHead == NULL){return -1;}
    (newHead)->value = value;
    (newHead)->next = *x;
    *x = newHead;
    return(0);
  }
  
  int pop(stack **x, char *value){
    if(x == NULL){return(-1);}
    if(*x == NULL){return(-1);}
    if(value == NULL){return(-1);}
    *value = (*x) -> value;
    *x = (*x) -> next;
    return(0);
  }
